### PR TITLE
Db 803 submission status

### PR DESCRIPTION
**Pull request by jworcestBAH**
Submission status now distinguishes between submissions with errors and those without.

### CODE REVIEW

File: tests/fileTests.py [hunk 1]
```diff
@@ -284,6 +284,21 @@ def test_check_status(self):
...
+        self.assertEqual(response.status_code, 200, msg=str(response.json))
+        self.assertEqual(
+            response.headers.get("Content-Type"), "application/json")
+        json = response.json
+        errorReportSub = None
+        for submission in json["submissions"]:
+            if submission["submission_id"] == self.error_report_submission_id:
+                errorReportSub = submission
+        self.assertIsNotNone(errorReportSub)
+        self.assertEqual(errorReportSub["status"], "validation_successful")
```
Comment (bsweger) [inline tests/fileTests.py]: do we need to do any checking around the new `validation_errors` as well?

**Comment by bsweger** [comment]
@jworcestBAH Done reviewing, and everything looks good (just one small unit test question).

Thinking ahead to the tech debt change I'm doing to capture total rows w/ errors in the job table...would it make sense as part of that for me to revisit jobTrackerInterface.py and use the new error count on the job table to set the new `validation_errors` and `validation_successful` statuses? That we we wouldn't have to pass around the errorDb interface. Though I suppose that wouldn't catch type errors.

One last thought--is there anything here that will need revisited once our warnings are in place? For example, if a job has warnings but no errors, do we need another status? If so, should we make a story for that so we don't forget?

File: tests/fileTests.py (hunk 1 above)
Comment (jworcestBAH) [inline tests/fileTests.py]: Expanded the list_submissions test to check that

**Comment by jworcestBAH** [comment]
The list submission setup function will need to specify severity when warnings are introduced.  From conversations so far, I don't think the frontend will display a separate submission status for warnings, but if this is desired we can add a call to the number of warnings function at that point.  
For switching to the error count on the job table, if that's not going to include all error types we should probably stick with this function so we don't get another scenario with the status saying no errors and a nonzero error count next to it.

**Comment by bsweger** [comment]
@jworcestBAH thanks for expanded the test. looks good--merging!

**Pull request merged**